File: backend/Models/General.cs
using System;
using System.Collections.Generic;

namespace backend.Models
{
    public partial class General
    {
        public int BookId { get; set; }
        public int AuthorId { get; set; }
        public int EditorialId { get; set; }
        public string? Title { get; set; }
        public int? Pages { get; set; }
        public string? Description { get; set; }
        public string Name { get; set; } = null!;
        public string Lastname { get; set; } = null!;
        public string Editorialname { get; set; } = null!;
        public string Location { get; set; } = null!;
    }
}

File: frontend/src/app/models/general.ts
import { Book } from "./book";

export class General implements Book{
    bookId?: number | undefined;
    title: string;
    description: string;
    pages: number;
    editorialId: number;
    authorId?: number;
    name?: string;
    lastname?: string;
    editorialname?: string;
    location? : string;

    constructor(bookId: number, title: string, description: string, pages: number, editorialId: number){
        this.bookId = bookId;
        this.description = description;
        this.pages = pages;
        this.editorialId = editorialId;
        this.title = title;
    }
}
